website/content/en/blog/2024-06-idle-power-matters-kepler-metrics-for-public-cloud-energy-efficiency.md

```diff
@@ -0,0 +1,150 @@
+
+## The Challenges of Estimating Idle Power
+
+Now that we understand the importance of isolating and reporting idle power, let's delve into how Kepler addresses the challenge of estimating idle power, especially in cloud environments where certain information, such as the number of co-located virtual machines, may be obscured.
+
+
+On bare metal, we can estimate idle power by identifying the minimum point of power consumption when server utilization is close to zero [2, 3]. However, we plan to extend this model with regression analysis to find the idle power also in busy systems.
+
+
+However, estimating idle power becomes more complex in virtual private clouds, where physical specifications and the number of sharing tenants (virtual machines) are not readily available. To address this, we've developed a training framework to extract and train only the dynamic power model, thereby reducing dependency on unknown factors and providing energy-efficient optimization for virtual machines [[4]](https://www.computer.org/csdl/proceedings-article/mascots/2023/10387542/1TKR5vzaWXe).
+
+At the beginning, as addressed in our previous blog post [[5]](https://www.cncf.io/blog/2023/10/11/exploring-keplers-potentials-unveiling-cloud-application-power-consumption/), we decided to disable the idle power of the container running on top of VM. Yet, we still need to tackle the idle power attribution and accounting issue. To strike a balance, we're developing an approach to estimate the idle power of virtual machines based on limited knowledge, specifically in scenarios where (i) physical cores are pinned to virtual cores without overprovisioning, and (ii) the same CPU architecture is emulated in the virtual machine. In such cases, we can apply a power model trained on the same or similar CPU architecture and estimate the idle power specific to the virtual machine based on the number of cores. For instance, if we have a power model trained for 64 cores with an idle power of 160W, a virtual machine provided with 8 virtual cores would account for 20W (i.e. 160/8).
+
+However, it's important to note that cloud waste, as illustrated in situation II of the example, may still not be fully accounted for at the user level without knowledge of the requested resources of all sharing tenants.
+
+## Conclusion
+
+This blog post underscores the importance of transparent, fine-grained energy metrics for public cloud efficiency. By addressing challenges such as idle power attribution and estimation, tools like Kepler pave the way for sustainable and energy-efficient cloud computing practices. However, it's crucial to acknowledge that complete waste accounting at the user level remains a challenge, necessitating further research and collaboration within the industry.
```

Comment:
@sunya-ch, great blog. QQ: If this blog is about creating awareness, is there an explicit CTA? Are there links to the community channel/communication channels where interested folks can collaborate or learn more?

Reply:
That's very good point. I will add CTA to the blog.